src/api/data-migration/version-3-to-version-4/update-applications/update-declaration-modern-slaveries/create-declaration-modern-slavery-version-rows.ts

```diff
@@ -0,0 +1,57 @@
+import { Connection } from 'mysql2/promise';
+import { DECLARATIONS } from '../../../../constants';
+import getAllDeclarationModernSlaveries from '../../../helpers/get-all-declaration-modern-slaveries';
+import createCuid from '../../../helpers/create-cuid';
+import executeSqlQuery from '../../../execute-sql-query';
+import { ApplicationDeclarationModernSlavery } from '../../../../types';
+
+const { WILL_ADHERE_TO_ALL_REQUIREMENTS, HAS_NO_OFFENSES_OR_INVESTIGATIONS, IS_NOT_AWARE_OF_EXISTING_SLAVERY } =
+  DECLARATIONS.LATEST_MODERN_SLAVERY_DECLARATIONS;
+
+/**
+ * createDeclarationModernSlaveryVersionRows
+ * For each modern slavery, create a new row in the DeclarationModernSlaveryVersion table.
+ * @param {Connection} connection: SQL database connection
+ * @returns {Promise<Array<object>>} executeSqlQuery response
+ */
+const createDeclarationModernSlaveryVersionRows = async (connection: Connection) => {
+  const loggingMessage = 'Creating modern slavery version entries with modern slavery relationships';
+
+  console.info('✅ %s', loggingMessage);
+
+  try {
+    const modernSlaveries = await getAllDeclarationModernSlaveries(connection);
+
+    if (!modernSlaveries.length) {
```

Comment:
should we not do the same !modernSlaveries || !modernSlaveries.length as other places?

Reply:
Nope, we have that check for "declarations for in progress applications", not "modern slaveries", as it works differently to the former.

This check (`getAllDeclarationModernSlaveries`); will always return an array.

Whereas the `getAllDeclarationsNonSubmittedApplications` check could return `false` or an array.